### ISSUE TITLE
? for docstring doesn't work after expression

### ISSUE THREAD

**Issue by andreasnoack** [JuliaLang/IJulia.jl#495]
See screenshot below

<img width="845" alt="screen shot 2017-01-04 at 10 47 34 pm" src="https://cloud.githubusercontent.com/assets/505001/21668435/d4013da8-d2cf-11e6-9ed0-09e77b757f07.png">

cc: @alanedelman

**Comment by yuyichao** [comment]
Seems reasonable? Should be documented if not already.

**Comment by andreasnoack** [comment]
I think it has caused some confusion. Usually, cells return the result of the last line. Would it have undesirable consequences to allow the second version to return the same as the first?

**Comment by shorty66** [comment]
symbols for special REPL-modes need to be placed at the start of a cell. 

These won't work:
```
#comment
?foo
```

```
a=1
;cd folder/
```

So this isn't really a doc-issue, but rather an issue about special REPL-modes.
On the other hand, shell mode seems to be crippled in IJulia.

**Comment by stevengj** [comment]
(Note that you can also do `shift-tab` on a symbol to get a pop-up with the docstring.)

**Comment by andreasnoack** [comment]
That seems like the better solution to recommend for notebooks. Would this be easier to communicate if the `?symbol` help was not available in notebooks? To me, it seems mainly to make sense in the REPL.

**Comment by shorty66** [comment]
The popup window opened by `shift-tab` does not render markdown formatted text.  Also, it is a little small if want to know about more complex functions with many methods. 
For example, try reading the help on the Package `UnicodePlots` in the popup-window.

I'd rather keep the `?symbol` for those cases where a nicely rendered version is useful.

**Comment by stevengj** [comment]
Yes, I'd prefer to keep both methods.  Also, `?` is instantly familiar to people coming from the REPL (or Python).

@Carreau, is there any chance of getting markdown formatting in the pop-up help tooltips someday?  We are supplying a mime bundle, after all.

**Comment by yuyichao** [comment]
The `doc` label is for documenting the behavior. Not about showing document.

**Comment by Carreau** [comment]
> @Carreau, is there any chance of getting markdown formatting in the pop-up help tooltips someday? We are supplying a mime bundle, after all.

Likely yes. I would like to have that as well. right now unfortunatelty it does not seem the tooltip might not survive the transition to jupyterlab (see https://github.com/jupyterlab/jupyterlab/issues/714) and pitch in if you like (or do not) the tooltip. 

I'll add on my todo list to see how much refactor we need to get markdown rendered in the tooltip.
